File: app/src/free/java/com/kksionek/gdzietentramwaj/view/AdProvider.java
package com.kksionek.gdzietentramwaj.view;

import android.content.Context;
import android.location.Location;
import android.view.ViewGroup;

import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.AdView;
import com.google.android.gms.ads.MobileAds;
import com.kksionek.gdzietentramwaj.R;

public class AdProvider implements AdProviderInterface {

    private AdView mAdView;

    @Override
    public void initialize(Context context, String adMobAppId) {
        MobileAds.initialize(context, adMobAppId);
    }

    @Override
    public void showAd(ViewGroup adView) {
        mAdView = (AdView) adView;
    }

    @Override
    public void resume() {
        if (mAdView != null) {
            mAdView.resume();
        }
    }

    @Override
    public void pause() {
        if (mAdView != null) {
            mAdView.pause();
        }
    }

    @Override
    public void loadAd(Context context, Location location) {
        if (mAdView != null) {
            AdRequest adRequest = new AdRequest.Builder()
                    .addTestDevice(context.getString(R.string.adMobTestDeviceS7Edge))
                    .setLocation(location)
                    .build();
            mAdView.loadAd(adRequest);
        }
    }
}

File: app/src/free/java/com/kksionek/gdzietentramwaj/view/AdProvider.kt
package com.kksionek.gdzietentramwaj.view

import android.content.Context
import android.location.Location
import android.view.View.VISIBLE
import android.view.ViewGroup

import com.google.android.gms.ads.AdRequest
import com.google.android.gms.ads.AdView
import com.google.android.gms.ads.MobileAds
import com.kksionek.gdzietentramwaj.BuildConfig
import com.kksionek.gdzietentramwaj.R
import com.kksionek.gdzietentramwaj.map.view.AdProviderInterface

class AdProvider : AdProviderInterface {

    private var adView: AdView? = null

    override fun initialize(context: Context) {
        if (BuildConfig.DEBUG) {
            val requestConfiguration = MobileAds.getRequestConfiguration().toBuilder()
                .setTestDeviceIds(
                    listOf(
                        context.getString(R.string.adMobTestDeviceS7Edge),
                        context.getString(R.string.adMobTestDeviceS7Edge2)
                    )
                ).build()
            MobileAds.setRequestConfiguration(requestConfiguration)
        }
        MobileAds.initialize(context)
    }

    override fun showAd(adView: ViewGroup) {
        this.adView = adView as AdView
    }

    override fun resume() {
        adView?.resume()
    }

    override fun pause() {
        adView?.pause()
    }

    override fun loadAd(context: Context, location: Location) {
        adView?.let {
            it.visibility = VISIBLE
            val adRequest = AdRequest.Builder()
                .setLocation(location)
                .build()
            it.loadAd(adRequest)
        }
    }
}
